### ISSUE TITLE
Exception is thrown when accessing ExcelPackage.Workbook property due to defined names

### ISSUE THREAD

**Issue by akashkarov** [EPPlusSoftware/EPPlus#260]
The following error occurs when I'm trying to open a workbook.
```
Name contains invalid characters or is not valid.
   at OfficeOpenXml.ExcelNamedRangeCollection.Add(String Name, ExcelRangeBase Range)
   at OfficeOpenXml.ExcelWorkbook.GetDefinedNames()
   at OfficeOpenXml.ExcelPackage.get_Workbook()
```

There is a part of `<definedNames>` section listed below from a workbook.xml:
```
<definedNames>
		<definedName name="\C" localSheetId="29">#REF!</definedName>
		<definedName name="\C" localSheetId="30">#REF!</definedName>
		<definedName name="\C" localSheetId="31">#REF!</definedName>
                ...
                <definedName name="\c1" localSheetId="29">#REF!</definedName>
		<definedName name="\c1" localSheetId="30">#REF!</definedName>
		<definedName name="\c1" localSheetId="31">#REF!</definedName>
                ...
                <definedName name="\G" localSheetId="29">#REF!</definedName>
		<definedName name="\G" localSheetId="30">#REF!</definedName>
		<definedName name="\G" localSheetId="31">#REF!</definedName>
                ...
<definedNames>
```
The file can be opened successfully with Excel interop library. Unfortunately I don't know how it gets to this state, but name attributes seems to be valid according to the Excel rules for defined names.

**Comment by JanKallman** [comment]
\C and \G are not allowed defined names in the Excel GUI, but seems to be accepted once the worksheet is loaded. I'll see if i can remove the validation from the load of the worksheet.

**Comment by akashkarov** [comment]
I have manually removed invalid defined names from the workbook.xml and repacked it back to Excel file. Now it can be opened, but I got another error on package Save() call (I'm using EPPlus sources locally for debug):
```
The cache source is not a worksheet
   at OfficeOpenXml.Table.PivotTable.PivotTableCacheInternal.get_SourceRange() in EPPlus-develop\src\EPPlus\Table\PivotTable\PivotTableCacheInternal.cs:line 104
   at OfficeOpenXml.ExcelWorkbook.AddPivotTableCache(PivotTableCacheInternal cacheReference, Boolean createWorkbookElement) in EPPlus-develop\src\EPPlus\ExcelWorkbook.cs:line 1526
   at OfficeOpenXml.Table.PivotTable.ExcelPivotCacheDefinition..ctor(XmlNamespaceManager nsm, ExcelPivotTable pivotTable) in EPPlus-develop\src\EPPlus\Table\PivotTable\ExcelPivotCacheDefinition.cs:line 46
   at OfficeOpenXml.Table.PivotTable.ExcelPivotTable..ctor(ZipPackageRelationship rel, ExcelWorksheet sheet) in EPPlus-develop\src\EPPlus\Table\PivotTable\ExcelPivotTable.cs:line 50
   at OfficeOpenXml.Table.PivotTable.ExcelPivotTableCollection..ctor(ExcelWorksheet ws) in EPPlus-develop\src\EPPlus\Table\PivotTable\ExcelPivotTableCollection.cs:line 40
   at OfficeOpenXml.ExcelWorksheet.get_PivotTables() in EPPlus-develop\src\EPPlus\ExcelWorksheet.cs:line 3515
   at OfficeOpenXml.ExcelWorkbook.ReadAllPivotTables() in EPPlus-develop\src\EPPlus\ExcelWorkbook.cs:line 1614
   at OfficeOpenXml.ExcelWorksheet.get_PivotTables() in EPPlus-develop\src\EPPlus\ExcelWorksheet.cs:line 3516
   at OfficeOpenXml.ExcelStyles.UpdateXml() in EPPlus-develop\src\EPPlus\ExcelStyles.cs:line 780
   at OfficeOpenXml.ExcelWorkbook.Save() in EPPlus-develop\src\EPPlus\ExcelWorkbook.cs:line 1143
   at OfficeOpenXml.ExcelPackage.Save() in EPPlus-develop\src\EPPlus\ExcelPackage.cs:line 845
```
Could it be related?

**Comment by JanKallman** [comment]
I dont think so. Can you attach you workbook?

**Comment by akashkarov** [comment]
Unfortunately, I cannot, it contains sensitive data. Is there anything else I can help with?

**Comment by JanKallman** [comment]
This should be solved in version 5.5

**Comment by vkirienko** [comment]
Could you please point to specific commit that resolves it?

**Comment by akashkarov** [comment]
I have checked v5.5 and reproduced the original issue again for the same workbook. Can you please refer to the changes made regarding defined names validation?

**Comment by JanKallman** [comment]
You should be able to use the latest commit in the develop branch or the release/epplus5.5.0 branch.
I made a fix for the message shown, but it it does not help I will need a workbook to reproduce this issue. Be aware that EPPlus does not support updating pivot tables with external sources. How ever we will try to retain them.

**Comment by akashkarov** [comment]
Please see a demo project with excel file inside where you can reproduce this issue.
[DefinedNamesTest.zip](https://github.com/EPPlusSoftware/EPPlus/files/5755540/DefinedNamesTest.zip)

**Comment by JanKallman** [comment]
Added a fix for validation of defined names 982e3b746337db0397817174b82c7cfeb4167d1f. Note that these names are not accepted inside of Excel and will still be restricted if added by the `Names.Add` methods in EPPlus. Loading of workbooks containing these names will however be permitted.

**Comment by JanKallman** [comment]
The "solved" comment above refered to the issue with the pivot table cache sources. Is that solved for you?

**Comment by JanKallman** [comment]
Fixed in version 5.5.1

**Comment by akashkarov** [comment]
I have checked version 5.5.1and the original issue is gone. But there is another one, when I'm trying to save the package with these defined names. Saving works ok, but then Excel treats the file as corrupted and offers recovering. After recovery these defined names are back in workbook.xml. Please see attached test project reproducing this issue.
[DefinedNamesTest_2.zip](https://github.com/EPPlusSoftware/EPPlus/files/5803214/DefinedNamesTest_2.zip)
